### PR TITLE
Log+filter adaption by type instead of id

### PR DESCRIPTION
**Pull request by fredriksvendsen**

### CODE REVIEW

File: backend/src/backend/filterlog.clj [hunk 1]
```diff
@@ -15,58 +15,74 @@
...
-
-  (if-not (= device_id nil)
-    (do (if (true? firstParam)
-          (do (def newstr (str " device_id IN (" device_id ")"))
-              (def firstParam false))
-          (def newstr (str " AND device_id IN (" device_id ")")))
-        (def querystr (str querystr newstr))))
+(defn adaption-filter
+  [adaption_type queryStr]
+  (if-not (= nil adaption_type)
```
Comment (skykanin) [inline backend/src/backend/filterlog.clj:23]: nil is false, you don't need to explicitly check if something is equal nil

File: backend/src/backend/filterlog.clj [hunk 2]
```diff
@@ -15,58 +15,74 @@
...
-    (do (if (true? firstParam)
-          (do (def newstr (str " system_log.adaption_id IN (" adaption_id ")"))
-              (def firstParam false))
-          (def newstr (str " AND system_log.adaption_id IN (" adaption_id ")")))
-        (def querystr (str querystr newstr))))
+(defn device-filter
+  [device_id queryStr]
+  (if-not (= nil device_id)
+    (let [basicQueryString "SELECT system_log.*, adaption.adaption_type, adaption.adaption_id
+  FROM system_log INNER JOIN adaption ON adaption.adaption_id = system_log.adaption_id"]
```
Comment (skykanin) [inline backend/src/backend/filterlog.clj:35]: this let binding formatting looks weird, remember to run `lein cljfmt fix` or even better add it as a pre-commit hook

Comment (fredriksvendsen) [reply backend/src/backend/filterlog.clj:35]: i did and tried running it again, still didn't fix these bindings.. fixed manually

**Review by ghost — APPROVED** [review]
Frontend seems to work perfectly now

**Pull request merged**